tests/cases/astype.py

```diff
@@ -0,0 +1,103 @@
+from .provider_test import ProviderTest
+from gunpowder import *
+import numpy as np
+
+class AsTypeTestSource(BatchProvider):
+
+    def setup(self):
+
+        self.provides(
+            ArrayKeys.RAW,
+            ArraySpec(
+                roi=Roi((0, 0, 0), (1000, 1000, 1000)),
+                voxel_size=(4, 4, 4)))
+
+        self.provides(
+            ArrayKeys.GT_LABELS,
+            ArraySpec(
+                roi=Roi((0, 0, 0), (1000, 1000, 1000)),
+                voxel_size=(4, 4, 4)))
+
+    def provide(self, request):
+
+        batch = Batch()
+
+        # have the pixels encode their position
+        for (array_key, spec) in request.array_specs.items():
+
+            roi = spec.roi
+
+            for d in range(3):
```

Comment:
I think these asserts are redundant. Gunpowder should make sure that any request that is made is valid according to the `ArraySpec` for any `ArrayKey` that is provided by this node.

Reply:
Potentially. This was included following [the example present in the pre-existing Downsample test](https://github.com/funkey/gunpowder/blob/445d2ed875e2d33de28ff786c9a9767ea84049f9/tests/cases/downsample.py#L30).

It seems like there is no harm in checking, but I will be happy to remove it if you think it muddies it. Thanks!

Reply:
Definitely not a big deal. I think we're good to merge with or without.
I'd vote for deleting it. Especially since it is already redundant in that test as well.